osquery/tables/system/linux/deb_packages.cpp

```diff
@@ -56,59 +70,65 @@ QueryData genDebPackagesImpl(QueryContext& context, Logger& logger) {
   // Drop to 'nobody' to ensure that the libdpkg library
   // can't change the package database. Privileges will be
   // restored automatically when the `dropper` object goes
-  // out of scope.
+  // out of scope.  In the meantime we'll buffer all logs
+  // since the reduced permissions will block creation
+  // of the log file.
   //
   // Note that this is *NOT* a security feature
-  auto dropper = DropPrivileges::get();
-  dropper->dropTo("nobody");
-
   QueryData results;
-
-  for (const auto& admindir : admindir_list) {
-    if (!pathExists(admindir).ok()) {
-      continue;
-    }
-    auto dpkg_query_exp = IDpkgQuery::create(admindir);
-    if (dpkg_query_exp.isError()) {
-      logError(logger,
-               "Failed to open the dpkg database",
-               dpkg_query_exp.takeError(),
-               admindir);
-
-      continue;
-    }
-
-    auto dpkg_query = dpkg_query_exp.take();
-
-    auto package_list_exp = dpkg_query->getPackageList();
-    if (package_list_exp.isError()) {
-      logError(logger,
-               "Failed to list the packages",
-               package_list_exp.takeError(),
-               admindir);
-
-      continue;
+  {
```

Comment:
Note the addition of a local scope around the block where we drop privileges.  Once this scope ends [the privileges are restored](https://github.com/sgress454/osquery/blob/0e130a865f96211b24b68b6ac3c425d7b6809596/osquery/core/system.cpp#L390-L406), so we can flush the error log safely.